dags/copy_deduplicate.py

```diff
@@ -101,13 +111,27 @@
         owner="ssuh@mozilla.com",
         email=["telemetry-alerts@mozilla.com", "ssuh@mozilla.com"],
         dag=dag,
-        arguments=('--schema_update_option=ALLOW_FIELD_ADDITION',),
+        arguments=("--schema_update_option=ALLOW_FIELD_ADDITION",),
     )
 
     copy_deduplicate_main_ping >> bq_main_events
 
     # Daily and last seen views on top of every Glean application.
 
+    telemetry_derived__core_clients_first_seen__v1 = bigquery_etl_query(
```

Comment:
It's awkward that this query is scheduled explicitly here rather than being generated in `bqetl_core`. I can understand that it's desirable to have it here in order to express the future dependency of baseline_clients_first_seen on it.

The "right" thing to do here would be to refactor all of this out of copy_deduplicate. In that scenario, `core_clients_first_seen` would be in `bqetl_core` and we would create a separate DAG in this repo for `glean_usage` that would run at 02:30 and contain the baseline first_seen, daily, and last_seen tasks.

I think that's probably a distraction at this point, but we should add a code comment here to explain the potentially surprising layout.

Reply:
If I'm understanding correctly, there would be three DAGs involved with the ETL:

* copy_deduplicate (scheduled at 01:00 UTC), containing live -> stable table queries
* bqetl_core (scheduled at 02:00ish UTC), containing ETL for the core ping
* glean_usage (scheduled 02:30 UTC), containing ETL for glean usage stuff

Sensors or offsetting queries is one way to go, but I think it makes sense for all of the dependencies to be in one place in the meanwhile. I would actually be very interested in scheduling the generated queries explicitly instead of via a wrapper script, but I don't know how that would affect airflow performance.

Reply:
> I would actually be very interested in scheduling the generated queries explicitly instead of via a wrapper script, but I don't know how that would affect airflow performance.

I agree this would be ideal, but I also worry about how this might affect Airflow performance. It might be fine. If we want to seriously pursue that path, we should consult with @haroldwoo.

> If I'm understanding correctly, there would be three DAGs involved with the ETL

Yep, your details match what I was thinking.

> Sensors or offsetting queries is one way to go, but I think it makes sense for all of the dependencies to be in one place in the meanwhile

Yes, I think the best thing to do at the moment is document why this query is included in this DAG.